x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/ConnectorSearchResult.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.application.connector;
+
+import org.elasticsearch.common.bytes.BytesReference;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.xcontent.ToXContentObject;
+import org.elasticsearch.xcontent.XContentBuilder;
+import org.elasticsearch.xpack.application.connector.syncjob.ConnectorSyncJob;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Represents the outcome of a search query in the connectors and sync job index, encapsulating the search result.
+ * It includes a raw byte reference to the result which can be deserialized into a {@link Connector} or {@link ConnectorSyncJob} object,
+ * and a result map for returning the data without strict deserialization.
+ */
+public class ConnectorSearchResult implements Writeable, ToXContentObject {
```

Comment:
Maybe we could name this `ConnectorsAPISearchResult`, otherwise one could think that this only returns instances of `Connector` and not `ConnectorSyncJob`.

We could also create two subclasses `ConnectorSearchResult` and `ConnectorSyncJobSearchResult`, which both extend from this class simply for the sake of naming.

Reply:
> Maybe we could name this ConnectorsAPISearchResult, otherwise one could think that this only returns instances of Connector and not ConnectorSyncJob.

@timgrein Could we go with this pattern and not do two subclasses, with builder pattern the inheritance is a bit messy as super `Builder` class will be constructing always the `ConnectorsAPISearchResult` class instead of the child classes